storage/netapp_e/na_eseries_storage_system.py

```diff
@@ -0,0 +1,319 @@
+
+    try:
+        raw_data = r.read()
+        if raw_data:
+            data = json.loads(raw_data)
+        else:
+            raw_data = None
+    except:
+        if ignore_errors:
+            pass
+        else:
+            raise Exception(raw_data)
+
+    resp_code = r.getcode()
+
+    if resp_code >= 400 and not ignore_errors:
+        raise Exception(resp_code, data)
+    else:
+        return resp_code, data
+
+
+def main():
+    argument_spec = basic_auth_argument_spec()
+    argument_spec.update(dict(
+        state=dict(required=True, choices=['present', 'absent']),
+        ssid=dict(required=True, type='str'),
+        controller_addresses=dict(type='list'),
+        array_wwn=dict(required=False, type='str'),
+        array_password=dict(required=False, type='str', no_log=True),
+        array_status_timeout_sec=dict(default=60, type='int'),
+        enable_trace=dict(default=False, type='bool'),
+        meta_tags=dict(type='list')
+    ))
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[('state', 'present', ['controller_addresses'])]
+    )
+
+    p = module.params
+
+    state = p['state']
+    ssid = p['ssid']
+    controller_addresses = p['controller_addresses']
+    array_wwn = p['array_wwn']
+    array_password = p['array_password']
+    array_status_timeout_sec = p['array_status_timeout_sec']
+    validate_certs = p['validate_certs']
+    meta_tags = p['meta_tags']
+    enable_trace = p['enable_trace']
+
+    try:
+        api_usr = p['api_username']
+        api_pwd = p['api_password']
+        api_url = p['api_url']
+    except KeyError:
+        api_usr = os.environ['API_USERNAME']
+        api_pwd = os.environ['API_PASSWORD']
+        api_url = os.environ['API_URL']
+
+    if not api_usr or not api_pwd or not api_url:
+        module.fail_json(msg="You must pass in api_username "
+                             "and api_password and api_url to the module. Alternatively "
+                             "you can specify API_URL, API_USERNAME and API_PASSWORD "
+                             "as environment variables.")
+
+    changed = False
+    array_exists = False
+
+    try:
+        (rc, resp) = request(api_url + "/storage-systems/{0}".format(ssid), headers=dict(Accept="application/json"),
+                             url_username=api_usr, url_password=api_pwd, validate_certs=validate_certs,
+                             ignore_errors=True)
+    except:
+        err = get_exception()
+        module.fail_json(msg="Error accessing storage-system with id [{0}]. Error [{1}]".format(ssid, str(err)))
+
+    if rc == 200:
+        array_exists = True
+        array_detail = resp
+
+        if state == 'absent':
+            changed = True
+            array_exists = False
+        elif state == 'present':
+            current_addresses = set(
+                [i for i in [array_detail['ip1'], array_detail['ip2']] if i is not None and len(i) > 1])
```

Comment:
If valid values of array_detail['ip*'] are strings or None than this can be simplified like:
``` python

frozenset(i for i in (array_detail['ip1'], array_detail['ip2']) if i)
```